src/main/java/practice/StreamPractice.java

```diff
@@ -1,80 +1,72 @@
 package practice;
 
-import java.util.Collections;
+import java.util.Arrays;
 import java.util.List;
+import java.util.function.IntUnaryOperator;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
 import model.Candidate;
+import model.Cat;
 import model.People;
 
 public class StreamPractice {
-    /**
-     * Given list of strings where each element contains 1+ numbers:
-     * input = {"5,30,100", "0,22,7", ...}
-     * return min integer value. One more thing - we're interested in even numbers.
-     * If there is no needed data throw RuntimeException with message
-     * "Can't get min value from list: < Here is our input 'numbers' >"
-     */
+    private static final String EXCEPTION_MESSAGE = "Can't get min value from list: ";
+
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        return numbers.stream()
+                .map(s -> s.split(","))
+                .flatMap(Arrays::stream)
+                .mapToInt(Integer::parseInt)
+                .filter(n -> n % 2 == 0)
+                .min()
+                .orElseThrow(() -> new RuntimeException(EXCEPTION_MESSAGE + numbers));
     }
 
-    /**
-     * Given a List of Integer numbers,
-     * return the average of all odd numbers from the list or throw NoSuchElementException.
-     * But before that subtract 1 from each element on an odd position (having the odd index).
-     */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        IntUnaryOperator oddSubtraction = (i) -> i % 2 != 0
+                ? (Integer) numbers.toArray()[i] - 1 : (Integer) numbers.toArray()[i];
+        return Stream.of(IntStream.range(0, numbers.size())
+                .map(oddSubtraction))
+                .flatMapToInt(n -> n)
+                .filter(n -> n % 2 != 0)
+                .average().getAsDouble();
     }
 
-    /**
-     * Given a List of `People` instances (having `name`, `age` and `sex` fields),
-     * for example, `Arrays.asList( new People(«Victor», 16, Sex.MAN),
-     * new People(«Helen», 42, Sex.WOMEN))`,
-     * select from the List only men whose age is from `fromAge` to `toAge` inclusively.
-     * <p>
-     * Example: select men who can be recruited to army (from 18 to 27 years old inclusively).
-     */
     public List<People> selectMenByAge(List<People> peopleList, int fromAge, int toAge) {
-        return Collections.emptyList();
+        Predicate<People> peoplePredicate = (people) -> people.getSex() == People.Sex.MAN
```

Comment:
let's use more informative namings for each predicate in your code
```suggestion
        Predicate<People> peoplePredicate = person -> person.getSex() == People.Sex.MAN
```